csharp-advanced-topics/Delegates/DelegateTests/DelegateUnitTests.cs

```diff
@@ -0,0 +1,18 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Delegates;
+namespace DelegateTests
+{
+    [TestClass]
+    public class DelegateUnitTests
+    {
+        [TestMethod]
+        public void TestCalcProduct()
+        {
+            int first = 20;
+            int second = 20;
+            int expected = 400;
+            int actual = Program.CalcProduct(first, second);
+            Assert.AreEqual(expected, actual);
+        }
+    }
+}
```

Comment:
Some tests seem to be missing. Why haven't you created two more tests to test the Action and Func respectively? I know tests don't make too much sense in this article, but we need to see how you deal with them since every article should have at least some tests. Tests are important for us since that way we can make sure source code works properly by running dotnet test in Jenkins. We also ensure that the results won't change when we bump the framework versions in the future.